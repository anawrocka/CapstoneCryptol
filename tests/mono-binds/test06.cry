module test06 where

test : {a} a -> a
test a = bar
  where
  foo : a
  foo = zero

  bar = foo
