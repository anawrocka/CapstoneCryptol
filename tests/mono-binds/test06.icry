:set debug=on
:set mono-binds=off
:l test06.cry
:set mono-binds=on
:l test06.cry
