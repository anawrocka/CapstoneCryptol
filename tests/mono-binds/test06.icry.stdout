Loading module Cryptol
Loading module Cryptol
Loading module test06
module test06
import Cryptol
/* Not recursive */
test06::test : {a} a -> a
test06::test = \{a} (a : a) ->
               bar
               where
                 /* Not recursive */
                 foo : a
                 foo = zero a
                 
                 /* Not recursive */
                 bar : a
                 bar = foo
                 
               

Loading module Cryptol
Loading module test06
module test06
import Cryptol
/* Not recursive */
test06::test : {a} a -> a
test06::test = \{a} (a : a) ->
               bar
               where
                 /* Not recursive */
                 foo : a
                 foo = zero a
                 
                 /* Not recursive */
                 bar : a
                 bar = foo
                 
               

